torch/fx/graph.py

```diff
@@ -356,8 +356,13 @@ def __deepcopy__(self, memo=None) -> 'Graph':
         """
         memo = memo if memo else {}
         g = Graph()
-        output_val = g.graph_copy(self, val_map=memo)
-        g.output(output_val)
+        output_vals = g.graph_copy(self, val_map=memo, return_output_node=True)
+        assert isinstance(output_vals, tuple)
+        output_val, old_output_val = output_vals
+        if hasattr(old_output_val, 'type'):
+            g.output(output_val, old_output_val.type)
+        else:
+            g.output(output_val)
```

Comment:
You can replace this whole `if/else` with `g.output(output_val, type_expr=getattr(old_output_val, 'type', None))